subgraph-js/src/schemaExtensions.ts

```diff
@@ -15,3 +15,9 @@ declare module 'graphql/type/definition' {
     resolveReference?: GraphQLReferenceResolver<TContext>;
   }
 }
+
+export function hasReferenceResolver(
```

Comment:
Perhaps this could return `type is GraphQLObjectType & Required<Pick<GraphQLObjectType, 'resolveReference'>>` or something (perhaps giving that type a name, and checking that the field is truthy) so that you won't need the `!` after calling it?

Reply:
Good suggestion: 97f0281